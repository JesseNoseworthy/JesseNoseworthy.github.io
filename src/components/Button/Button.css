@import '../../styles/variables.css';
@import '../../styles/mixins.css';

:local(.Button) {
  cursor: pointer;
  transition: $animationSpeed;
}

:local(.AnimatingButton) {
  @media (--desktopAll) {
    transition: $animationSpeed;
    transform: perspective(1px) translateZ(0);
    position: relative;
    &:before {
      content: '';
      position: absolute;
      z-index: -1;
      top: 0;
      left: 0;
      bottom: 0;
      right: 0;
      transform-origin: 50% 0;
      transition: $animationSpeed;
    }
    &:not(:hover) {
      &:before {
        transform: scaleY(0);
      }
    }
    &:hover {
       &:before {
        transform: scaleY(1);
      }
    }
  }
}

:local(.AnimatingButton-business),
:local(.AnimatingButton-consumer) {
  &:before {
    background: $businessHeader-SecondaryBackground;
  }
}

:local(.AnimatingButton-pro-photo-video) {
  &:before {
    background: $pro-photo-video-Header-SecondaryBackground;
  }
}