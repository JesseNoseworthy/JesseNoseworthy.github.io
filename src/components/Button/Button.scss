// All Button styles go here

button {
  width: 130px;
  height: 45px;
  border-radius: 15px;
}