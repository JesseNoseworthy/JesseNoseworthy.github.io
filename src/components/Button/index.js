import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import styles from './Button.css';

const Button = ({
  className,
  children,
  handleClick,
  isActive,
  isAnimating,
  currentSegment,
  ...others
}) => (
  <button
    className={classNames(
      styles.Button,className,
      {[styles.AnimatingButton]: isAnimating},
      {[styles[`AnimatingButton-${currentSegment}`]]: isAnimating}
    )} {...others}>
    {children}
  </button>
);

Button.propTypes = {
  className: PropTypes.string,
  children: PropTypes.any,
  isAnimating: PropTypes.bool,
  handleClick: PropTypes.func,
  currentSegment: PropTypes.string,
  isActive: PropTypes.bool
};

Button.defaultProps = {
  isAnimating: false
}

export default Button;