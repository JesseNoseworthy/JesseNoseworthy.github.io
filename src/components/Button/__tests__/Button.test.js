import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import Button from '../index';

describe('Button', () => {
  describe('Snapshot tests', () => {
    it('Basic render', () => {
      const component = renderer.create(
        <Button />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    describe('Shallow rendering (no children)...', () => {
      it('produces a `button` element type', function() {
        expect(shallow(<Button />).type()).toEqual('button');
      });
    });

    describe('Mounting to the virtual DOM...', () => {
      it('allows us to set props', () => {
        const wrapper = mount(<Button type="submit" />);
        expect(wrapper.props().type).toEqual('submit');
        wrapper.setProps({ type: 'submit' });
        expect(wrapper.props().type).toEqual('submit');
      });

      it('responds to click events', () => {
        const handleClick = jasmine.createSpy();
        const wrapper = mount(<Button onClick={handleClick} />);
        wrapper.find('button').simulate('click');
        expect(handleClick).toHaveBeenCalled();
      });
    });

    it('should keep arbitrary attributes not designated as component props', () => {
      const button = render(<Button aria-label="test" />).find('button');

      const actual = button.attr('aria-label');
      const expected = 'test';

      expect(actual).toEqual(expected);
    });
 it('populated state', () => {
      const component = renderer.create(
        <Button 
          className={'sample-button'}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
