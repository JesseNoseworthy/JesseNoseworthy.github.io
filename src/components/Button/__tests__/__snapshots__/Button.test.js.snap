exports[`Button Snapshot tests Basic render 1`] = `
<button
  className="" />
`;

exports[`Button Snapshot tests populated state 1`] = `
<button
  className="sample-button" />
`;
